### PR TITLE
Expose the environment timestamp to the agent

### PR DESCRIPTION
**Pull request by diegoferigo**
This PR extends the interface of the environment exposed to the agent / user adding a `timestamp` method. It is useful e.g. for logging or plotting.

### CODE REVIEW

File: ignition/src/GazeboWrapper.cpp [hunk 1]
```diff
@@ -384,6 +384,19 @@ bool GazeboWrapper::initialized()
...
+{
+    auto dt = pImpl->gazebo.physics.maxStepSize;
+    auto iterationCount = pImpl->gazebo.server->IterationCount();
+
+    if (!iterationCount || !pImpl->gazebo.server) {
+        gymppError << "Failed to get the simulated time" << std::endl;
+        return 0.0;
+    }
+
+    return dt * iterationCount.value();
```
Comment (traversaro) [inline ignition/src/GazeboWrapper.cpp]: This is not related to the PR as it is a limitation of the Gazebo Server API, but these seems error prone, as it would break if at some point Gazebo changes the step size during a simulation (just wanted to comment this, even if nothing can be changed in this PR).

File: gym_ignition/runtimes/pybullet_runtime.py [hunk 2]
```diff
@@ -214,6 +222,9 @@ def step(self, action: Action) -> State:
         if not self.action_space.contains(action):
             logger.warn("The action does not belong to the action space")
 
+        # Increase the counter
+        self.counter += 1
```
Comment (traversaro) [inline gym_ignition/runtimes/pybullet_runtime.py]: Small suggestion: can we store the timestamp directly, and increase it by `1 / self.agent_rate` at every step? I don't think it will happen soon, but in this way the method will continue to work even if the agent_rate changes during the simulation.

File: gym_ignition/base/runtime.py [hunk 3]
```diff
@@ -50,3 +50,16 @@ def unwrapped(self):
...
+    def timestamp(self) -> float:
+        """
+        Return the timestamp associated to the execution of the environment.
+
+        In real-time environments, the timestamp is the time read from the host system.
+        In simulated environments, the timestamp is the simulated time, which might not
+        match the real-time in the case of RTF different than 1.
+
+        Returns:
+            A float indicating the current environment timestamp
```
Comment (traversaro) [inline gym_ignition/base/runtime.py]: What are the units of the timestamp?

Comment (diegoferigo) [reply gym_ignition/base/runtime.py]: At this early stage I didn't want to put any constraints. It is implementation dependent, think about gazebo and yarp for instance.

Comment (traversaro) [reply gym_ignition/base/runtime.py]: Ok, even if I think that both in YARP and in Gazebo the unit of the timestamps is units (when expressed as a scalar).

Comment (diegoferigo) [reply gym_ignition/base/runtime.py]: The timestamp could be an incremental counter, a time that starts from zero, or an unix timestamp for instance. All these are just a scalar representing a timestamp.

File: gym_ignition/runtimes/pybullet_runtime.py (hunk 2 above)
Comment (diegoferigo) [reply gym_ignition/runtimes/pybullet_runtime.py]: Sure, even better :+1:

File: ignition/src/GazeboWrapper.cpp (hunk 1 above)
Comment (diegoferigo) [reply ignition/src/GazeboWrapper.cpp]: Good point, thanks! I think we can just leave a comment in the code as a reminder

**Comment by diegoferigo** [comment]
@traversaro you can find the diff wrt your previous review [here](https://github.com/robotology/gym-ignition/compare/5bc507e535a6a137d9c36da05475b106eeace90c..5afa9592e7a8998e5ce402c81049d13e29def2ae)
